latch_cli/docker_utils/__init__.py

```diff
@@ -0,0 +1,197 @@
+import functools
+from dataclasses import dataclass
+from enum import Enum, auto
+from pathlib import Path
+from textwrap import dedent
+from typing import List
+
+import yaml
+
+from latch_cli.types import LatchWorkflowConfig
+
+print = functools.partial(print, flush=True)
+
+
+class DockerCmdBlockOrder(Enum):
+    FIRST = auto()
+    PRECOPY = auto()
+    POSTCOPY = auto()
+    LAST = auto()
+
+
+@dataclass
+class DockerCmdBlock:
+    comment: str
+    commands: List[str]
+    order: DockerCmdBlockOrder
+
+
+def get_prologue(config: LatchWorkflowConfig) -> List[str]:
+    return [
+        "# latch base image + dependencies for latch SDK -- removing these will break the workflow",
+        f"from {config.base_image}",
+        f"run python3 -m pip install latch=={config.latch_version}",
+    ]
+
+
+def get_epilogue() -> List[str]:
+    return [
+        "# latch internal tagging system + expected root directory -- changing these lines will break the workflow",
+        "arg tag",
+        "env FLYTE_INTERNAL_IMAGE $tag",
+        "WORKDIR /root",
+    ]
+
+
+def infer_commands(pkg_root: Path) -> List[DockerCmdBlock]:
+    commands: List[DockerCmdBlock] = []
+
+    # apt requirements in workflow directory
+    if (pkg_root / "requirements.apt").exists():
+        print("Install apt requirements from `requirements.apt`")
```

Comment:
```suggestion
        print("Installing apt requirements from `requirements.apt`")
```

Reply:
I disagree -- this is declaring what the Dockerfile will do not what it is doing currently

Reply:
yeah but I also don't really understand why you're printing this here anyway

maybe `print("Adding apt requirements installation phase")`